### ISSUE TITLE
[META] Crowdsourced endpoint and property descriptions

### ISSUE THREAD

**Issue by CarbonAlabel** [esi/esi-issues#471]
Many of the endpoint and property descriptions in the ESI specification are nondescript, incorrect, or leave out information which might be required for someone with little understanding of EVE's game mechanics to properly utilize them.

It might be worth looking into a scheme similar to the moderately successful [third-party dev docs](https://github.com/ccpgames/eveonline-third-party-documentation), that would allow more experienced EVE third-party devs to enhance the existing descriptions by creating pull requests.

**Comment by Robbilie** [comment]
we could use githubs wiki…? :)

**Comment by deadlybulb** [comment]
I like the idea, I'm not sure how feasible it would be depending on what you mean.  I'm assuming you'd want to crowdsource the descriptions in the swagger.json spec.  It looks like some of these descriptions are auto-generated (e.g. list of alternate routes).  Also, if I remember correctly, the swagger.json they end up delivering is actually assembled automatically from several independent annotated endpoints.

I can think of at least two possible ways forward:

- ask to embed a URL in each endpoint description to some third-party dev docs like site
- ask that their process to autogenerate the docs sources some part of the description field from a github project that takes PRs.

**Comment by CarbonAlabel** [comment]
> Also, if I remember correctly, the swagger.json they end up delivering is actually assembled automatically from several independent annotated endpoints.

That is true. For reference, [here](https://github.com/ccpgames/esi-routes/blob/master/esi_routes/web.py) is how it is done in the only ESI endpoint which has been published so far.
Moving those YAML strings from the endpoints' source code to files in a public repository would be one way of accomplishing what I'm suggesting, as well as providing a much more elegant alternative to https://github.com/xxpizzaxx/esi-archive.

If people were to start regularly creating issues here about lacking descriptions, it would be borderline spam, not to mention a pretty big waste of developer time to personally fix them all. Approving PRs made by other people (I'm guessing there would be a decent amount of people willing to make those) should be much faster, so I'd like to hear if CCP devs would consider implementing such a process to be worth their time.

**Comment by fc-pinky** [comment]
We very much welcome contributions to the developers website to expand on unclear fields, routes, and mechanics! There have been more than a few contributions there already.

As nobody replied to this ticket in the last few years, I am going to close it up, under the assumption that everyone said what they wanted to say here :) Feel free to start a discussion about this on Discord, or make PRs to https://github.com/esi/esi-docs :)
